scripts/next_step_assignment_sample_receipt.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+from EPPs.common import StepEPP
+from pyclarity_lims.entities import Protocol
+
+
+class AssignNextStepSampleReceipt(StepEPP):
+    """
+    This script checks to see if any of the relevant step UDFs are answered indicating that a manager review is required
+    """
+
+    def _run(self):
+
+        # obtain the actions of the step then creates a StepActions entity for the current step
+        actions = self.process.step.actions
+
+        # obtain the next actions in the step then creates a list of dict for next_actions for the step
+        next_actions = actions.next_actions
+
+        # check to see if review argument flag is present, if 'true' then set all next actions to "review"
+        if self.process.udf['Dry ice remaining in package?'] =='No' or self.process.udf['Container(s) undamaged and sealed?'] == 'No'\
+            or self.process.udf['Samples frozen?'] == 'No' or self.process.udf['>= 50ul sample present in wells or tubes?']=='No':
+            # for all artifacts in next_actions update the action to "review"
+            for next_action in next_actions:
+                next_action['action'] = 'review'
+
+        else:
+            current_step = self.process.step.configuration  # configuration gives the ProtocolStep entity.
```

Comment:
This `else` section looks very generic and could be used (or is used?) elsewhere.
It would be worth refactoring this as a generic function `next_step_or_complete`

Reply:
Added to StepEPP in common